src/components/internal/Button/DownloadButton/DownloadButton.tsx

```diff
@@ -69,21 +79,33 @@ function DownloadButton({ className, disabled, endpointName, params, setError, e
         onDownloadRequested?.();
     };
 
+    const getButtonLabel = useCallback(() => (isFetching ? `${i18n.get('downloading')}..` : i18n.get('download')), [isFetching]);
+    const getButtonIcon = useCallback(() => (isFetching ? <Spinner size={'small'} /> : <Download />), [isFetching]);
+
     return (
-        <div className="adyen-pe-download">
+        <div
+            className={classNames('adyen-pe-download', {
+                'adyen-pe-download-button-container': !iconButton,
+                'adyen-pe-download-icon-button-container': iconButton,
+            })}
+        >
             {isSmContainer ? (
                 <Button iconButton={true} variant={ButtonVariant.TERTIARY} onClick={onClick}>
                     {isFetching ? <Spinner size={'small'} /> : <Download />}
                 </Button>
             ) : (
                 <Button
-                    className={classNames('adyen-pe-download__button', { 'adyen-pe-download__button--loading': isFetching }, className)}
+                    className={classNames(
+                        'adyen-pe-download__button',
+                        { 'adyen-pe-download__button--loading': isFetching, 'adyen-pe-download__button--icon': iconButton },
+                        className
+                    )}
                     disabled={disabled || isFetching}
-                    variant={ButtonVariant.SECONDARY}
+                    variant={iconButton ? ButtonVariant.TERTIARY : ButtonVariant.SECONDARY}
                     onClick={onClick}
-                    iconLeft={isFetching ? <Spinner size={'small'} /> : <Download />}
+                    {...(!iconButton && { iconLeft: isFetching ? <Spinner size={'small'} /> : <Download /> })}
```

Comment:
This looks like `getButtonIcon()` to me. Is that the intent here? If so, then here is a suggestion:
```suggestion
                    {...(!iconButton && { iconLeft: getButtonIcon() })}
```